testing/src/main/scala/sbt/JUnitXmlTestsListener.scala

```diff
@@ -31,13 +31,14 @@ import sbt.protocol.testing.TestResult
 /**
  * A tests listener that outputs the results it receives in junit xml
  * report format.
- * @param outputDir path to the dir in which a folder with results is generated
+ * @param targetDir directory in which test reports are generated
  */
-class JUnitXmlTestsListener(val outputDir: String, legacyTestReport: Boolean, logger: Logger)
+class JUnitXmlTestsListener(val targetDir: File, legacyTestReport: Boolean, logger: Logger)
```

Comment:
If the current API is using String to pass the absolute path around, then we should keep that behavior to minimize the change in case this listener class is extended by others.

Reply:
Good point! I've updated to add (another!) legacy constructor which keeps the old API.

Reply:
(To be clear: the current API accepts `outputDir` as a String in the constructor, but actually writes to `targetDir` which is constructed as `${outputDir}/test-reports`. I believe the new code follows this behaviour as closely as possible)